scrapy/contrib/downloadermiddleware/redirect_offsite.py

```diff
@@ -0,0 +1,34 @@
+import re
+from scrapy import log
+from scrapy.http import Request
+from scrapy.exceptions import IgnoreRequest
+from scrapy.utils.httpobj import urlparse_cached
+from scrapy.contrib.downloadermiddleware.redirect import RedirectMiddleware
+
+
+class RedirectOffsiteMiddleware(RedirectMiddleware):
+
+    def process_response(self, request, response, spider):
+        redirected = super(RedirectOffsiteMiddleware, self).process_response(request, response, spider)
+        return self.offsite_filtered(redirected, spider)
+
+    def offsite_filtered(self, redirected, spider):
+
+        allowed_domains = getattr(spider, 'allowed_domains', None)
+
+        if not allowed_domains:
+            return redirected
+
+        if isinstance(redirected, Request):
```

Comment:
This method is called from `process_response`, how can the redirected ever be a `Request` here?